### PR TITLE
get rid of `no_alloc` feature, add `alloc` feature

### PR DESCRIPTION
**Pull request by FrozenDroid**
I believe that this is the proper way to handle optional alloc/std features.
When using a crate with no default features enabled, it should assume the lowest common denominator. So, no allocator, no std. This PR makes that work.
Maybe this warrants a minor version increment, since we're changing the features?

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
This is going to have to wait for a minor version release, because although I do agree this is a better API decision, I do have to wait for Nom and any other dependent libraries to have time to increment any dependencies. I was actually considering implementing this myself, but this more than welcome and exactly how I would have implemented this. It does not look like this will break Nom, however, which is good, since Nom hardcods the minimal-lexical version. Once this is merged, I'll open an issue for a minor version increment.

File: Cargo.toml
Comment (Alexhuszagh) [inline Cargo.toml:30]: This needs an additional minor version increment. The version should go to `0.2`, due to backwards-incompatible API changes.

**Review by Alexhuszagh — CHANGES_REQUESTED** [review]
Just needs a minor version increment, otherwise, the changes look excellent.

File: Cargo.toml
Comment (FrozenDroid) [reply Cargo.toml:30]: I agree.

Comment (Alexhuszagh) [reply Cargo.toml:30]: Push the relevant commit changing the version (you can merge the commits or not, either is fine) and I'll approve and merge the PR.

**Comment by Alexhuszagh** [comment]
Merged, and the new version has been published.

**Pull request merged**
